bitcoin/src/psbt/mod.rs

```diff
@@ -121,8 +179,98 @@ impl Psbt {
         Ok(psbt)
     }
 
-    /// Extracts the `Transaction` from a PSBT by filling in the available signature information.
-    pub fn extract_tx(self) -> Transaction {
+    #[inline]
+    fn get_input_total_value(&self) -> Option<u64> {
+        let mut input_amount = 0;
+        for (vin, psbtin) in self.unsigned_tx.input.iter().zip(self.inputs.iter()) {
+            let prev_vout = vin.previous_output.vout;
+
+            // Return early if any input is missing both utxo types.
+            let value = psbtin.witness_utxo.as_ref().map(|o| o.value.to_sat()).or_else(|| {
+                psbtin
+                    .non_witness_utxo
+                    .as_ref()
+                    .and_then(|tx| tx.output.get(prev_vout as usize))
+                    .map(|o| o.value.to_sat())
+            })?;
+
+            input_amount += value;
+        }
+
+        Some(input_amount)
+    }
+
+    #[inline]
+    fn get_output_total_value(&self) -> u64 {
+        self.unsigned_tx.output.iter().map(|o| o.value.to_sat()).sum::<u64>()
+    }
+
+    #[inline]
+    fn get_fee(&self) -> GetFeeResult {
+        let input_value = self.get_input_total_value();
+        let output_value = self.get_output_total_value();
+        match input_value.map(|i| i.checked_sub(output_value)) {
+            Some(Some(fee)) => GetFeeResult::Ok { fee, output_value },
+            Some(None) => GetFeeResult::UnderFlow,
+            None => GetFeeResult::MissingInputValue,
+        }
+    }
+
+    /// The default `max_fee_rate` value used for extracting transactions with [`extract_tx`]
+    ///
+    /// As of 2023, even the biggest overpayers during the highest fee markets only paid around
+    /// 1000 sats/vByte. 25k sats/vByte is obviously a mistake at this point.
+    ///
+    /// [`extract_tx`]: Psbt::extract_tx
+    pub const DEFAULT_MAX_FEE_RATE: f64 = 25000.0_f64;
```

Comment:
Any reason not to use the `FeeRate` type?

Reply:
Not particularly.

Reply:
Cool, lets use it then, floats are the worst.